src/Algebra/Ordered/Bundles.agda

```diff
@@ -0,0 +1,346 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions of ordered algebraic structures like promonoids and
+-- posemigroups (packed in records together with sets, orders,
+-- operations, etc.)
+------------------------------------------------------------------------
+
+-- The contents of this module should be accessed via `Algebra.Order`.
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Ordered.Bundles where
+
+open import Algebra.Core
+open import Algebra.Bundles
+open import Algebra.Ordered.Structures
+open import Level using (suc; _⊔_)
+open import Relation.Binary using (Rel)
+open import Relation.Binary.Bundles using (Preorder; Poset)
+
+------------------------------------------------------------------------
+-- Bundles of preordered structures
+
+-- Preordered magmas (promagmas)
+
+record Promagma c ℓ₁ ℓ₂ : Set (suc (c ⊔ ℓ₁ ⊔ ℓ₂)) where
```

Comment:
I'm mildly worried about the “pro” prefix because I've seen “proset” used to mean a partial order where we don't have/never talk about equality. I don't know whether this is standard, but it's very relevant to the setoid setting.

Reply:
Hum, indeed. I thought I had seen at least promonoid somewhere, but I might be fooling myself. Staying consistent with the literature is important though, so I'm open to alternative suggestions. Then there's the issue of keeping names short 😅